packages/gatsby-plugin-utils/src/polyfill-remote-file/types.ts

```diff
@@ -93,3 +93,33 @@ export function isImage(node: {
 
   return node.mimeType.startsWith(`image/`) && node.mimeType !== `image/svg+xml`
 }
+
+export type ImageCdnTransformArgs = WidthOrHeight & {
+  format: string
+  cropFocus?: ImageCropFocus | Array<ImageCropFocus>
+  quality: number
+}
+
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type ImageCdnSourceImage = {
+  url: string
+  mimeType: string
+  filename: string
+  internal: { contentDigest: string }
+}
+
+/**
+ * The function is used to optimize image delivery by generating URLs that leverage CDN capabilities
+ * @param {ImageCdnSourceImage} source - An object representing the source image, including properties like
+ * URL, filename, and MIME type.
+ * @param {ImageCdnTransformArgs} imageArgs - An object containing arguments for image transformation, such as
+ * format, quality, and crop focus.
+ * @param {string} pathPrefix - A string representing the path prefix to be prepended to the
+ * generated URL.
+ * @returns {string} A string representing the generated URL for the image on the CDN.
```

Comment:
Maybe we can add more nuance here about relative/absolute urls - for now we only use relative one which will be on same domain as the site, but technically (at least conceptually, would need to test that) it could also be absolute url

Absolute one potentially could be used for example for user managed `imgix` instance (kind of like https://nextjs.org/docs/app/api-reference/next-config-js/images#imgix for Next) - tho for now there is no easy way for user to define something like (tho they could be hacky way of setting `global.__GATSBY.imageCDNUrlGeneratorModulePath` in `gatsby-config/gatsby-node` (definitely would not recommend using that as that's internal thing and might break at any point)

Reply:
I didn't yet do any testing here, but https://github.com/gatsbyjs/gatsby/pull/37932 somewhat implies that returning absolute urls is fine

Reply:
Did my testing and providing absolute urls works fine in the pipeline. When different domain is used there would probably be questions about things like `preconnect`, but this is out of scope of current work, so will update the comment here that either relative (same domain) or absolute urls can be provided